e2e/cypress/support/index.ts

```diff
@@ -15,19 +15,64 @@
 
 // Import commands.js using ES2015 syntax:
 import './commands';
+import '../integration/constants';
 
 // Alternatively you can use CommonJS syntax:
 // require('./commands')
 
+const IAMV2 = 'v2.1';
+const IAMV1 = 'v1';
+
 before(function () {
-  if (!Cypress.env('ADMIN_TOKEN')) {
+  if (!Cypress.env('IAM_VERSION')) {
+    // tslint:disable-next-line:no-string-throw
+    throw('MISSING ENVIRONMENT VARIABLE: ' +
+      `You must pass CYPRESS_IAM_VERSION. Must be one of: "${IAMV2}", "${IAMV1}".`);
+  }
+
+  if (Cypress.env('IAM_VERSION') !== IAMV2 && Cypress.env('IAM_VERSION') !== IAMV1) {
+    // tslint:disable-next-line:no-string-throw
+    throw('INCORRECT ENVIRONMENT VARIABLE: ' +
+      `You must pass CYPRESS_IAM_VERSION. Must be one of: "${IAMV2}", "${IAMV1}". ` +
+      `You passed: ${Cypress.env('IAM_VERSION')}`);
+  }
+
+  if (!Cypress.env('RUN_FLAKY')) {
+    // tslint:disable-next-line:no-string-throw
+    throw('MISSING ENVIRONMENT VARIABLE: You must pass CYPRESS_RUN_FLAKY.' +
+      'Must be one of: "yes", "no".');
+  }
+
+  if (Cypress.env('RUN_FLAKY') !== 'yes' && Cypress.env('RUN_FLAKY') !== 'no') {
+    // tslint:disable-next-line:no-string-throw
+    throw('INCORRECT ENVIRONMENT VARIABLE: You must pass CYPRESS_RUN_FLAKY. ' +
+      'Must be one of: "yes", "no".' +
+      `You passed: ${Cypress.env('RUN_FLAKY')}`);
+  }
+
     cy.adminLogin('/').then(() => {
       const admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'));
-      cy.generateAdminToken(admin.id_token);
+
+      // this needs to use the id_token because generateAdminToken will
+      // throw a confusing error if you are on the wrong version of IAM so we need
+      // to do this first.
+      cy.request({
+        auth: { bearer: admin.id_token },
+        url: '/apis/iam/v2beta/policy_version'
+      }).then((response) => {
+        if (Cypress.env('IAM_VERSION') === IAMV2) {
+          expect(response.body.version.major).to.equal('V2');
+        } else {
+          expect(response.body.version.major).to.equal('V1');
+        }
+      });
```

Comment:
`A test asserts that the version returned from the API matches CYPRESS_IAM_VERSION` This is covered here, correct?

Reply:
oops missed that, my b